libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1146,6 +1345,27 @@ string YulUtilFunctions::conversionFunction(Type const& _from, Type const& _to)
 			solUnimplemented("Fixed point types not implemented.");
 			break;
 		case Type::Category::Array:
+			if (_from == _to)
+			{
+				body = "converted := value";
+				break;
+			}
+			else
+			{
+				ArrayType const& from = dynamic_cast<decltype(from)>(_from);
+				ArrayType const& to   = dynamic_cast<decltype(to)>  (_to);
+
+				ReferenceType const& fromNoPtr =
+					*TypeProvider::withLocation(&from, from.location(), false);
+				ReferenceType const& toNoPtr =
+					*TypeProvider::withLocation(&to, to.location(), false);
+
+				if (fromNoPtr == toNoPtr)
+				{
+					body = "converted := value";
+					break;
+				}
+			}
 			solUnimplementedAssert(false, "Array conversion not implemented.");
```

Comment:
Shouldn't that just go into the else part of the if from line 1363?

Reply:
(and then remove the `break` statements)

Reply:
Well, logically it wouldn't make a difference.. If none of the two branches cause a break it's an unimplemented/unhandled situation..

Reply:
I think it is rather confusing to use break in switch like that.